swift/aperture/main.swift

```diff
@@ -7,27 +7,31 @@ func quit(_: Int32) {
 }
 
 func record() throws {
-  let destinationPath = CommandLine.arguments[1]
-  let fps = CommandLine.arguments[2]
-  let cropArea = CommandLine.arguments[3]
-  let showCursor = CommandLine.arguments[4] == "true"
-  let highlightClicks = CommandLine.arguments[5] == "true"
-  let displayId = CommandLine.arguments[6] == "main" ? CGMainDisplayID() : UInt32(CommandLine.arguments[6])
-  let audioDeviceId = CommandLine.arguments[7]
-
-  var coordinates = [String]()
+  // TODO: Use JSON and `Codable` here when Swift 4 is out
+
+  let args = CommandLine.arguments
+  let destination = args[1]
+  let fps = args[2]
+  let cropArea = args[3]
+  let showCursor = args[4]
+  let highlightClicks = args[5]
+  let displayId = args[6]
+  let audioDeviceId = args[7]
+
+  var cropRect: CGRect?
   if cropArea != "none" {
-    coordinates = CommandLine.arguments[3].components(separatedBy: ":")
+    let points = cropArea.components(separatedBy: ":").map { CGFloat(Double($0)!) }
```

Comment:
CGRect has an initilizer for Doubles so you don't have to do the extra cast to CGFloat